spdmlib/src/message/encapsulated.rs

```diff
@@ -0,0 +1,163 @@
+        Ok(cnt)
+    }
+
+    fn spdm_read(
+        _context: &mut common::SpdmContext,
+        r: &mut Reader,
+    ) -> Option<SpdmGetEncapsulatedRequestPayload> {
+        u8::read(r)?; // param1
+        u8::read(r)?; // param2
+
+        Some(SpdmGetEncapsulatedRequestPayload {})
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct SpdmEncapsulatedRequestPayload {
+    pub request_id: u8,
+}
+
+impl SpdmCodec for SpdmEncapsulatedRequestPayload {
+    fn spdm_encode(
+        &self,
+        _context: &mut common::SpdmContext,
+        bytes: &mut Writer,
+    ) -> Result<usize, SpdmStatus> {
+        let mut cnt = 0usize;
+        cnt += self
+            .request_id
+            .encode(bytes)
+            .map_err(|_| SPDM_STATUS_BUFFER_FULL)?; // param1
+        cnt += 0u8.encode(bytes).map_err(|_| SPDM_STATUS_BUFFER_FULL)?; // param2
+        Ok(cnt)
+    }
+
+    fn spdm_read(
+        _context: &mut common::SpdmContext,
+        r: &mut Reader,
+    ) -> Option<SpdmEncapsulatedRequestPayload> {
+        let request_id = u8::read(r)?; // param1
+        u8::read(r)?; // param2
+
+        Some(SpdmEncapsulatedRequestPayload { request_id })
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct SpdmDeliverEncapsulatedResponsePayload {
+    pub request_id: u8,
+}
+
+impl SpdmCodec for SpdmDeliverEncapsulatedResponsePayload {
+    fn spdm_encode(
+        &self,
+        _context: &mut common::SpdmContext,
+        bytes: &mut Writer,
+    ) -> Result<usize, SpdmStatus> {
+        let mut cnt = 0usize;
+        cnt += self
+            .request_id
+            .encode(bytes)
+            .map_err(|_| SPDM_STATUS_BUFFER_FULL)?; // param1
+        cnt += 0u8.encode(bytes).map_err(|_| SPDM_STATUS_BUFFER_FULL)?; // param2
+        Ok(cnt)
+    }
+
+    fn spdm_read(
+        _context: &mut common::SpdmContext,
+        r: &mut Reader,
+    ) -> Option<SpdmDeliverEncapsulatedResponsePayload> {
+        let request_id = u8::read(r)?; // param1
+        u8::read(r)?; // param2
+
+        Some(SpdmDeliverEncapsulatedResponsePayload { request_id })
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct SpdmEncapsulatedResponseAckPayload {
+    pub request_id: u8,
+    pub payload_type: SpdmEncapsulatedResponseAckPayloadType,
+    pub ack_request_id: u8,
+}
+
+impl SpdmCodec for SpdmEncapsulatedResponseAckPayload {
+    fn spdm_encode(
+        &self,
+        _context: &mut common::SpdmContext,
+        bytes: &mut Writer,
+    ) -> Result<usize, SpdmStatus> {
+        let mut cnt = 0usize;
+        cnt += self
+            .request_id
+            .encode(bytes)
+            .map_err(|_| SPDM_STATUS_BUFFER_FULL)?; // param1
+        cnt += self
+            .payload_type
+            .encode(bytes)
+            .map_err(|_| SPDM_STATUS_BUFFER_FULL)?; // param2
+        cnt += self
+            .ack_request_id
+            .encode(bytes)
+            .map_err(|_| SPDM_STATUS_BUFFER_FULL)?;
```

Comment:
This field is only available in SPDM 1.2. Not in SPDM 1.1

Please add SPDM version check.